OSDOCS-12183: adding how to delete manifests (#83246)
openshift/openshift-docs: modules/microshift-examples-of-usecase.adoc

// Module included in the following assemblies:
//
// * microshift//running_applications/microshift-deleting-resource-manifests.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-examples-of-usecase_{context}"]
= Use cases for resource manifest deletion

The following explains the use case in which the resource manifest deletion is used.

[id="data-removal-scenario_{context}"]

> == Data removal scenario

Reviewer: ```suggestion
== Manifest removal scenario
```

Author: done

Use the following procedure in the data removal scenario to completely delete the resource defined in the kustomize manifest file.

.Procedure

. Identify the file that needs to be placed in the `delete` subdirectories.
. Move the manifest file into one of the `delete` subdirectories  under the configured manifests directory by running the following command:
+
[source,terminal]
----

> [sudo] mv /usr/lib/microshift/manifests.d/some-user-manifest /usr/lib/microshift/manifests.d/delete/some-user-manifest

Reviewer: The `delete` subdir needs to be created first.
`sudo mkdir -p /usr/lib/microshift/manifests.d/delete`

Author: done

----

> . {microshift-short} detects and removes the resource after the manifest file is placed in the `delete` subdirectories.

Reviewer: MicroShift needs to be restarted to make it effective
`sudo systemctl restart microshift`

Author: done

[id="upgrade-scenario_{context}"]

> == Upgrade scenario

Reviewer: ```suggestion
== Manifest upgrade scenario
```

Author: done

Use the following procedure in the upgrade scenario where some resources need to be removed while retaining others to preserve data.

.Procedure

. Identify the manifest file that requires updating. It is not necessary to include the entire resource `spec` in the `delete` subdirectories.

> . Specify the metadata, such as `name` and `namespace` of the resource to be removed, in the `delete` subdirectories.

Reviewer: I would say:
- identify the manifest (without file, as it's most likely a directory)
- create new manifest for resource deletion that need to be deleted (`spec` can be skipped) in path like manifests.d/delete/my-manifest-delete/{kustomization.yaml,...}
- create new manifest to apply in manifests.d/my-manifest

Author: done

> . {microshift-short} detects and removes the specified object without affecting related data.

Reviewer: There's no automatic action, microshift needs to be restarted

// Module included in the following assemblies:
//
// * microshift//running_applications/microshift-deleting-resource-manifests.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-examples-of-usecase_{context}"]
= Use cases for resource manifest deletion

The following explains the use case in which the resource manifest deletion is used.

[id="data-removal-scenario_{context}"]
== Manifest removal scenario
Use the following procedure in the data removal scenario to completely delete the resource defined in the kustomize manifests.

.Procedure

. Identify the manifest that needs to be placed in the `delete` subdirectories.
. Create the `delete` subdirectory in which the manifest will be placed by running the following command:
+
[source,terminal]
----

> sudo mkdir -p _<path_of_deletefile>_ <1>

Author: done

----

> <1> Replace `_<path_of_deletefile>_` with the path of delete subdirectory such as `/usr/lib/microshift/manifests.d/delete`.

Author: done

. Move the manifest file into one of the `delete` subdirectories  under the configured manifests directory by running the following command:
+
[source,terminal]
----
[sudo] mv _<path_of_manifests>_ _<path_of_deletefile>_ <1>
----

> <1> Replace `_<path_of_manifests>_` with the path of manifest that has to be deleted such as `/usr/lib/microshift/manifests.d/some-user-manifest` and replace `_<path_of_deletefile>_` with the path of delete subdirectory such as `/usr/lib/microshift/manifests.d/delete`.

Reviewer: ```suggestion
<1> Replace `_<path_of_manifests>_` with the path of manifest that has to be deleted such as `/etc/microshift/manifests.d/some-user-manifest` and replace `_<path_of_deletefile>_` with the path of delete subdirectory such as `/etc/microshift/manifests.d/delete`.
```

Author: done

> . {microshift-short} detects and removes the resource after the manifest file is placed in the `delete` subdirectories.

Author: done

. Restart {microshift-short} by running the following command:
+
[source,terminal]
----
$ sudo systemctl restart microshift
----

[id="upgrade-scenario_{context}"]
== Manifest upgrade scenario
Use the following procedure in the upgrade scenario where some resources need to be removed while retaining others to preserve data.

.Procedure

. Identify the manifest that requires updating.
. Create new manifests for resource deletion and place it in the `delete` subdirectories path. It is not necessary to include the `spec` in these manifests.
. Create new manifests that needs to be applied in the manifests path.
. {microshift-short} detects and removes the specified object without affecting related data.

> . Restart {microshift-short} by running the following command:

Reviewer: Same here, I think we should merge them or swap them

Author: done

// Module included in the following assemblies:
//
// * microshift//running_applications/microshift-deleting-resource-manifests.adoc

:_mod-docs-content-type: CONCEPT
[id="microshift-examples-of-usecase_{context}"]
= Use cases for resource manifest deletion

> The following explains the use case in which the resource manifest deletion is used.

Author: Done. I have made changes in assembly and deleted this module
